### PR TITLE
fix crash caused by unicode chars

### PR DESCRIPTION
**Pull request by domisterwoozy**
Prevent a crash during lexing for unicode chars.

### CODE REVIEW

File: toolchain/lex/tokenized_buffer.cpp [hunk 1]
```diff
@@ -909,7 +909,8 @@ class [[clang::internal_linkage]] TokenizedBuffer::Lexer {
     // TODO: Need to add support for Unicode lexing.
     if (LLVM_LIKELY(position + 2 >= static_cast<ssize_t>(source_text.size()) ||
                     source_text[position + 1] != '#' ||
-                    !IsIdStartByteTable[source_text[position + 2]])) {
+                    !IsIdStartByteTable[static_cast<unsigned char>(
+                        source_text[position + 2])])) {
```
Comment (zygoloid) [inline toolchain/lex/tokenized_buffer.cpp]: It looks like we have the same error in `LexKeywordOrIdentifier` (L869), and in the macro-generated `Dispatch##LexMethod##SymbolToken` functions (L1075, `OneCharTokenKindTable`), though neither of those can ever actually see a negative `char` value. Would you mind changing those to cast to `unsigned char` too, for consistency?

**Review by zygoloid — COMMENTED** [review]
Thank you!

Can you also provide a test? (Eg, add a file to `toolchain/lex/testdata` with `r#\x80` or something like that in it.)

**Review by zygoloid — APPROVED** [review]
Perfect, thank you!

**Comment by zygoloid** [comment]
Argh, the relevant parts of `tokenized_buffer.cpp` just got moved to `lex.cpp`.

**Comment by chandlerc** [comment]
Thanks for updating, merging!

**Pull request merged**